### PR TITLE
Drop changes on finalize in postprocessor and user object

### PR DESCRIPTION
**Pull request by fdkong**
Refs #12948

### CODE REVIEW

File: test/tests/transfers/multiapp_conservative_transfer/tests [hunk 1]
```diff
@@ -4,6 +4,7 @@
   [./conservative_transfer]
     type = Exodiff
     input = master_conservative_transfer.i
+    max_parallel = 6
```
Comment (permcody) [inline test/tests/transfers/multiapp_conservative_transfer/tests]: This isn't the right fix - We should probably put the standard "errorIfParallel" stuff in this transfer until we can look at why it doesn't work for DistributedMesh. I'll accept this for now, but it's not clear to the user why you are limiting to 6 (other than you are hiding that it's actually broken).

**Review by permcody — DISMISSED** [review]

File: test/tests/transfers/multiapp_conservative_transfer/tests (hunk 1 above)
Comment (fdkong) [reply test/tests/transfers/multiapp_conservative_transfer/tests]: It is for the integration. I will fix it soon.

**Comment by fdkong** [comment]
@permcody  You may need to Review this PR again. I use `MultiAppMeshFunctionTransfer` instead of `MultiAppNearestNodeTransfer` in master_conservative_transfer.i

**Comment by fdkong** [comment]
`MultiAppNearestNodeTransfer` is unstable in parallel.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**